src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -202,6 +375,51 @@ object LoadVCF {
       .toArray
   }
 
+  // parses the Variant (key), leaves the rest to f
+  def parseLines[C](makeContext: () => C)(f: (C, String, RegionValueBuilder) => Unit)(
+    lines: RDD[WithContext[String]], t: Type): RDD[RegionValue] = {
+    lines.mapPartitions { it =>
+      new Iterator[RegionValue] {
+        val region = MemoryBuffer()
+        val rvb = new RegionValueBuilder(region)
+        val rv = RegionValue(region)
+
+        val context: C = makeContext()
+
+        var present: Boolean = false
+
+        def hasNext: Boolean = {
+          while (!present && it.hasNext) {
+            present = it.next().map { l =>
+              val vcfLine = new VCFLine(l)
+              region.clear()
+              rvb.start(t)
+              rvb.startStruct()
+              val b = vcfLine.parseAddVariant(rvb)
+              if (b) {
+                f(context, l, rvb)
+
+                rvb.endStruct()
+                rv.setOffset(rvb.end())
```

Comment:
just to clarify, this will always be 0 (for now), right?

Reply:
Yes, because we just cleared it, although this code shouldn't rely on it.  (For example: it might be nice to use 0 to represent missing for pointer types, array and binary.  Then RegionValueBuilder would start at offset 8.)